core/src/main/scala/org/apache/spark/sql/execution/joins/LocalJoin.scala

```diff
@@ -137,8 +160,22 @@ case class LocalJoin(leftKeys: Seq[Expression],
   // return empty here as code of required variables is explicitly instantiated
   override def usedInputs: AttributeSet = AttributeSet.empty
 
-  override def inputRDDs(): Seq[RDD[InternalRow]] =
-    streamedPlan.asInstanceOf[CodegenSupport].inputRDDs()
+  lazy val buildCodegenRDDs = buildPlan.asInstanceOf[CodegenSupport].inputRDDs()
+
+  override def inputRDDs(): Seq[RDD[InternalRow]] = {
+    // If the build side has shuffle dependencies.
+    if (buildCodegenRDDs.length == 1 && buildCodegenRDDs(0).dependencies.size >= 1 &&
```

Comment:
Can there be a case where buildCodegenRDDs are more than 1? I think we should just walk through all the dependencies of all buildCodegenRDDs to be on the safe side because now this is used for all kinds of joins. Earlier it was always on a replicated table so it was known that only one RDD will be coming in.